scripts/benchmarks_local.py

```diff
@@ -267,12 +311,111 @@ def generate_benchmark_ci_args(parsed_args: Namespace, specific_run_type: RunTyp
                 corerun_path = corerun_capture[0]
             bdn_args_unescaped += [ corerun_path ]
 
+    # for commit in all_commits: There is not a way to run multiple Wasm's at once via CI, instead will split single run vs multi-run scenarios
+    elif specific_run_type == RunType.WasmInterpreter:
+        raise TypeError("WasmInterpreter does not support combined benchmark ci arg generation, use single benchmark generation and loop the benchmark_ci.py calls.")
+
+    elif specific_run_type == RunType.WasmAOT:
+        raise TypeError("WasmInterpreter does not support combined benchmark ci arg generation, use single benchmark generation and loop the benchmark_ci.py calls.")
+
     if parsed_args.bdn_arguments:
         bdn_args_unescaped += [parsed_args.bdn_arguments]
     benchmark_ci_args += [f'--bdn-arguments={" ".join(bdn_args_unescaped)}']
     getLogger().info(f"Finished generating benchmark_ci.py arguments for {specific_run_type.name} run type using artifacts in {parsed_args.artifact_storage_path}.")
     return benchmark_ci_args
 
+def generate_single_benchmark_ci_args(parsed_args: Namespace, specific_run_type: RunType, commit: str) -> list:
+    getLogger().info(f"Generating benchmark_ci.py arguments for {specific_run_type.name} run type using artifacts in {parsed_args.artifact_storage_path}.")
+    benchmark_ci_args = []
+    bdn_args_unescaped = []
+    benchmark_ci_args += ['--architecture', parsed_args.architecture]
+    benchmark_ci_args += ['--frameworks', parsed_args.framework]
+    benchmark_ci_args += ['--filter', parsed_args.filter]
+    benchmark_ci_args += ['--dotnet-path', parsed_args.dotnet_dir_path]
+    benchmark_ci_args += ['--csproj', parsed_args.csproj]
+    benchmark_ci_args += ['--incremental', "no"]
+    benchmark_ci_args += ['--bdn-artifacts', os.path.join(parsed_args.artifact_storage_path, f"BenchmarkDotNet.Artifacts.{specific_run_type.name}.{commit}.{start_time.strftime('%y%m%d_%H%M%S')}")] # We add the commit hash to the artifact path because we are only running one commit at a time and they would clobber if running more than one commit perf type
+
+    if specific_run_type == RunType.CoreRun:
+        bdn_args_unescaped += [
+                                '--anyCategories', 'Libraries', 'Runtime',
+                                '--logBuildOutput',
+                                '--generateBinLog'
+                            ]
+        bdn_args_unescaped += [ '--corerun', os.path.join(get_run_artifact_path(parsed_args, RunType.CoreRun, commit), "Core_Root", f'corerun{".exe" if is_windows(parsed_args) else ""}') ]
```

Comment:
Is there a reason why some of these have space at the beginning and end?

Reply:
Fixed, along with some formatting.